plugins/API/RowEvolution.php

```diff
@@ -317,6 +317,13 @@ private function getRowEvolutionMetaData($idSite, $period, $date, $apiModule, $a
         $reportMetadata = API::getInstance()->getMetadata($idSite, $apiModule, $apiAction, $apiParameters, $language,
             $period, $date, $hideMetricsDoc = false, $showSubtableReports = true);
 
+        // ignore idGoal if no report can be found with it
+        if (empty($reportMetadata) && isset($apiParameters['idGoal'])) {
```

Comment:
if we go with this "hack", be good to leave a comment about it for what it fixes. Wonder if it could lead to some "wrong" results? Probably not? 

Did you have a chance to think about what a "proper" fix might look like? Can't think of anything right now apart from having another report instance of all the regular reports with a "idGoal" parameter but this might have other downsides? Like in `Reports.addReports` event we could maybe add for all "Goal" reports another report instance and set idGoal parameter? Might be just as well a different hack thought and bit more complicated but might enable then other features as well. Really just a random thought and not even sure if it is possible.

Reply:
> Did you have a chance to think about what a "proper" fix might look like?

No, but I don't think adding additional reports to the reports list would be good. Maybe instead we shouldn't pass a `idGoal` to the rowevolution in this case, but another param like `showGoalMetricsFor` or similar to be able to show other metrics within the same report?

Reply:
Passing idGoal if needed in rowevolution would be a really good solution I reckon, did not think of that. If it is easy possible 👍

Reply:
Will have a look as soon as I have some time. Maybe that would a better solution than going with the changes provided in this PR